staticjinja/dep_graph.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding:utf-8 -*-
+
+"""
+Dependency graph for staticjina
+"""
+
+from copy import deepcopy
+
+
+class DepGraph(object):
+    """
+    A directed graph which will handle dependencies between templates and data
+    files in a site.
+
+    :param site:
+        A :class:`Site <Site>` object.
+
+    """
+    def __init__(self, site):
+        self.parents = dict(
+                (s, set()) for s in site.jinja_names + site.data_names
+                )
+        self.children = deepcopy(self.parents)
+
+        for filename in site.jinja_names:
+            self.parents[filename] = site.get_file_dep(filename)
+            for d in self.parents[filename]:
+                self.children[d].add(filename)
```

Comment:
It would also make this class more generic since it would rely on site so heavily. If we could get of the use of site in update() it'd be completely generic.